treescript/src/treescript/script.py

```diff
@@ -51,31 +55,33 @@ async def do_actions(config, task, actions, repo_path):
         actions (list): the actions to perform
         repo_path (str): the source directory to use.
     """
-    await checkout_repo(config, task, repo_path)
+    repo_type = "hg"  # TODO parametrize when git support is added
+    vcs = get_vcs_module(repo_type)
+    await vcs.checkout_repo(config, task, repo_path)
 
     # Split the action selection up due to complexity in do_actions
     # caused by different push behaviour, and action return values.
     if "merge_day" in actions:
-        await perform_merge_actions(config, task, actions, repo_path)
+        await perform_merge_actions(config, task, actions, repo_path, repo_type)
         return
 
     num_changes = 0
     if "tag" in actions:
-        num_changes += await do_tagging(config, task, repo_path)
+        num_changes += await vcs.do_tagging(config, task, repo_path)
     if "version_bump" in actions:
-        num_changes += await bump_version(config, task, repo_path)
+        num_changes += await bump_version(config, task, repo_path, repo_type)
```

Comment:
`bump_version` and `l10n_bump` are in their own modules. That's why I'm passing `repo_type` down.